expr/boolean.go

```diff
@@ -411,62 +409,58 @@ func Contains(compare Boolean) Boolean {
 // See the comments of the various type implementations
 // of this method as some types limit the operand to equality and
 // the various types handle coercion in different ways.
-func Comparison(op string, primitive astzed.Primitive) (Boolean, error) {
+func Comparison(op string, val *zed.Value) (Boolean, error) {
 	// String literals inside Zed are parsed as zng bstrings
 	// (since bstrings can represent a wider range of values,
 	// specifically arrays of bytes that do not correspond to
 	// UTF-8 encoded strings).
-	if primitive.Type == "string" {
-		primitive = astzed.Primitive{Kind: "Primitive", Type: "bstring", Text: primitive.Text}
+	if val.Type == zed.TypeString {
+		val = zed.NewValue(zed.TypeBstring, val.Bytes)
 	}
```

Comment:
`zed.TypeOfString` and `zed.TypeOfBstring` are handled by the same case in the switch statement below, so you can delete all of this.